onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorEinSum.cpp

```diff
@@ -45,6 +45,28 @@ class DmlOperatorEinSum : public DmlOperator, public EinSumHelper
             }
             break;
 
+        case RecognizedOperatorType::OuterProduct:
+            {
+                std::array<uint32_t, 4> aSizes = {1, 1, m_inputTensorDescs[0].GetSizes().back(), 1};
```

Comment:
<strike>🤔 How does `outputDescs` pass the DML validator? If you're coercing the inputs to 4D (note DML hasn't required 4D coercion for a few years now, as the API directly accepts 2D), then doesn't the output need coercion to 4D too, because otherwise the API validation would complain about mismatching dimensions of 2D vs 4D?</strike>

[update] Ah, the `TensorDesc::TensorDesc`'s parameter `minDimensionCount` is still set to 4 - I've been meaning to remove that parameter for a while since it's 95% irrelevant now, given the DML API directly accepts any rank tensor. So, we should pass 2D for both inputs and outputs or explicit 4D for inputs and outputs. I don't care which, but I just don't want it to break later when we later remove `minDimensionCount` (which will simplify other code and less us delete a number of tedious axis adjustmentns).

Reply:
p.s. Remove the ampersand.

TensorDesc[[[[&]]]] bTensorDesc = TensorDesc(m_inputTensorDescs[1].GetDmlDataType(), bSizes);